src/category_theory/connected.lean

```diff
@@ -0,0 +1,374 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import category_theory.limits.shapes.pullbacks
+import category_theory.limits.shapes.binary_products
+import category_theory.limits.shapes.equalizers
+import category_theory.limits.preserves
+
+/-!
+# Connected category
+
+Define a connected category
```

Comment:
Should explain more about what this file is for.

1. explain the main definition
2. state the most interesting equivalences with other definitions
3. explain the theorem "The functor `(X × -)` preserves any connected limit." that you eventually prove.